packages/core/src/engine/EngineUtils.ts

```diff
@@ -201,6 +201,17 @@ function mapElem<T, S>(f: (arg: T) => S, e: Elem<T>): Elem<S> {
       tag: e.tag,
       contents: mapTup3((x) => mapTup2(f, x), e.contents),
     };
+  } else if (e.tag === "Arc") {
+    const l: [[S, S], [S, S, S], [S, S]] = [
+      // TODO sad :(
```

Comment:
The `map*` functions in this module seem to be very redundant and not easy to maintain. My understanding is that they are only here for converting autodiff numbers to JS numbers? I was wondering if there's a better way to unpack autodiff numbers so we can avoid boilerplate like this @hypotext? Probably some kind of dynamic dispatch?